Pet Skins and Ears

SabreCat opened the pull request:
Implements the following:
- A new set of always-available premium avatar skins based on eight of the basic pets.
- A new set of head accessories matched to those skins.
- The ability to display and purchase a set of gear items on the Avatar Customization page

In support of the above, Lodash was added to `middleware.js`.

Alys committed:
add message about Resting in the Inn to top of Dailies column when preferences.sleep is true

crookedneighbor committed:
Add alert box to dailies resting in the inn message

crookedneighbor committed:
Unpend in test, pend failing tests

crookedneighbor committed:
Correct bad tests in common suite

crookedneighbor committed:
Enable new inn behavior

SabreCat committed:
feat(customize): Animal skins and ears

Implements a base-pet-themed set of avatar skins and head accessories, the latter of which has additional handling for the user to purchase them from the Avatar Customization page.

SabreCat committed:
chore(grunt): compile sprites

crookedneighbor commented on website/src/middleware.js:
```diff
@@ -198,7 +198,8 @@ module.exports.locals = function(req, res, next) {
     tavern: tavern, // for world boss
-    worldDmg: (tavern && tavern.quest && tavern.quest.extra && tavern.quest.extra.worldDmg) || {}
+    worldDmg: (tavern && tavern.quest && tavern.quest.extra && tavern.quest.extra.worldDmg) || {},
+    _: _
```
Is this to provide lodash as a resource to something on the server? What?

SabreCat replied:
It's for this: https://github.com/SabreCat/habitrpg/blob/pet-skins/website/views/options/profile.jade#L7

That's what @lefnire and I came up with after banging around quite a bit, but if there's a smoother way to get this logic in without needing lodash as middleware, that may be preferable.

SabreCat committed:
chore(news): Bailey

hairlessbear commented:
@SabreCat This is incredibly minor, but is there any chance that you could add a credit to me in the Bailey announcement for the Inn improvements? I submitted the [Inn behavior change PR](https://github.com/HabitRPG/habitrpg/pull/5073) and collaborated with Blade on it.

crookedneighbor commented:
@SabreCat Yes, definitely do that.

SabreCat committed:
Merge remote-tracking branch 'origin/new_inn_behavior_PR' into pet-skins

SabreCat commented:
Aha! Absolutely, thanks.

SabreCat committed:
test(purchases): Gem buying tests

Also corrects the byline for Inn improvements to credit primary coder @hairlessbear.

SabreCat committed:
fix(test): Unlock takes query, not params

SabreCat committed:
fix(purchases): Handle undefined/null balance

SabreCat committed:
fix(unlocks): Handle undefined/null balance
